cmd/tarball.go

```diff
@@ -83,4 +86,53 @@ func runTarball(binariesLocation string) {
 	tar := fmt.Sprintf("%s.tar.gz", name)
 	fmt.Println(" >  ", tar)
 	sh.RunCommand("tar", "zcf", filepath.Join(prefix, tar), "-C", tmpDir, name)
+
+	// Windows systems don't have tar available by default. Produce archives in
+	// the common zip format additionally.
+	if goos == "windows" {
+		archive := name + ".zip"
+		fmt.Println(" >  ", archive)
+		if err := createZIP(filepath.Join(prefix, archive), dir); err != nil {
+			fatal(fmt.Errorf("Could not create ZIP archive: %w", err))
+		}
+	}
+}
+
+// createZIP creates a ZIP archive at the given path containing the specified
+// directory.
+func createZIP(path, dir string) error {
+	f, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	w := zip.NewWriter(f)
+	defer w.Close()
+
+	prefix := filepath.Dir(dir)
+	walker := func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+
+		r, err := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer r.Close()
+
+		name := strings.TrimLeft(strings.TrimPrefix(path, prefix), "/")
```

Comment:
Could we avoid this by passing both `prefix` and `archive` to createZIP instead of joining them ahead of time?

Reply:
These are different prefixes. The name of the file written to the zip archive needs to be made relative, it's not ensured that user provided path is relative though. By changing the current working directory first the walk function could be written differently. I don't see it'll result in simpler or less code.

Reply:
Makes sense. It's a bit hard to understand the intent just looking at the combination of string trimming functions.

Reply:
Fair enough. I don't have a great idea how to simplify it, and my motivation to invest more time for a niche Windows solution is low ;-)